default.nix

```
@@ -9,45 +9,76 @@ let
     inherit system;
   };
 
-  nix-dev = pkgs.stdenv.mkDerivation {
-    name = "nix-dev";
-    src = ./.;
-    nativeBuildInputs = with pkgs.python310.pkgs; [
-      linkify-it-py
-      myst-parser
-      sphinx
-      sphinx-book-theme
-      sphinx-copybutton
-      sphinx-design
-      sphinx-notfound-page
-      sphinx-sitemap
-    ];
-    buildPhase = ''
-      make html
-    '';
-    installPhase =
-      let
-        # Various versions of the Nix manuals, grep for (nix-manual)=
-        # FIXME: This requires human interaction to update!
-        # See ./CONTRIBUTING.md for details.
-        releases = [
-          "2.19"
-          "2.18"
-          "2.13"
-        ];
-        inputName = version: pkgs.lib.strings.replaceStrings [ "." ] [ "-" ] version;
-        src = version: (import inputs."nix_${inputName version}").default.doc;
-        copy = version: ''
-          cp -R ${src version}/share/doc/nix/manual/* $out/manual/nix/${version}
-        '';
-      in
-      with pkgs.lib.strings;
-      ''
-        mkdir -p $out/manual/nix/{${concatStringsSep "," releases}}
-        ${concatStringsSep "\n" (map copy releases)}
-        cp -R build/html/* $out/
+  nix-dev =
+    let
+      # Various versions of the Nix manuals, grep for (nix-manual)=
+      # FIXME: This requires human interaction to update!
+      # See ./CONTRIBUTING.md for details.
+      # Using Nix from Nixpkgs could simplify handling, but would requires a bit more setup here
+      releases = {
+        latest = "2.19";
+        rolling = "2.18";
+        stable = "2.18";
+        prev-stable = "2.13";
+      };
+      nixpkgs-releases = {
+        stable = "23.11";
+        prev-stable = "23.05";
+      };
+      inputName = version: pkgs.lib.strings.replaceStrings [ "." ] [ "-" ] version;
+      src = version: inputs."nix_${inputName version}";
+      manual = version: (import (src version)).default.doc;
+      copy = version: ''
+        cp -Rf ${manual version}/share/doc/nix/manual/* $out/manual/nix/${version}
       '';
-  };
+      redirects = version: ''
+        sed 's#^\(.*\) \(.*\) #/nix/manual/${version}\1 /nix/manual/${version}\2 #g' ${src version}/doc/manual/_redirects >> $out/_redirects
```

There seems to be a problem with this regex, it matches the lines with comments.

Something like this works:

```suggestion
        sed '/# /d;s#^\(.*\) \(.*\) #/nix/manual/${version}\1 /nix/manual/${version}\2 #g' ${src version}/doc/manual/_redirects >> $out/_redirects
```